subworkflows/local/vcf_annotate_all/main.nf

```diff
@@ -20,43 +20,43 @@ workflow VCF_ANNOTATE_ALL {
     vep_extra_files
 
     main:
-    ch_reports   = Channel.empty()
-    ch_vcf_ann   = Channel.empty()
-    ch_tab_ann   = Channel.empty()
-    ch_json_ann  = Channel.empty()
-    ch_versions  = Channel.empty()
+    reports = Channel.empty()
+    vcf_ann = Channel.empty()
+    tab_ann = Channel.empty()
+    json_ann = Channel.empty()
+    versions = Channel.empty()
 
     if (tools.split(',').contains('merge') || tools.split(',').contains('snpeff')) {
         VCF_ANNOTATE_SNPEFF(vcf, snpeff_db, snpeff_cache)
 
-        ch_reports  = ch_reports.mix(VCF_ANNOTATE_SNPEFF.out.reports)
-        ch_vcf_ann  = ch_vcf_ann.mix(VCF_ANNOTATE_SNPEFF.out.vcf_tbi)
-        ch_versions = ch_versions.mix(VCF_ANNOTATE_SNPEFF.out.versions.first())
+        reports = reports.mix(VCF_ANNOTATE_SNPEFF.out.reports)
+        vcf_ann = vcf_ann.mix(VCF_ANNOTATE_SNPEFF.out.vcf_tbi)
+        versions = versions.mix(VCF_ANNOTATE_SNPEFF.out.versions)
     }
 
     if (tools.split(',').contains('merge')) {
-        vcf_ann_for_merge = VCF_ANNOTATE_SNPEFF.out.vcf_tbi.map{ meta, vcf, tbi -> [meta, vcf] }
+        vcf_ann_for_merge = VCF_ANNOTATE_SNPEFF.out.vcf_tbi.map{ meta, vcf, tbi -> [ meta, vcf ] }
         VCF_ANNOTATE_MERGE(vcf_ann_for_merge, fasta, vep_genome, vep_species, vep_cache_version, vep_cache, vep_extra_files)
 
-        ch_reports  = ch_reports.mix(VCF_ANNOTATE_MERGE.out.reports)
-        ch_vcf_ann  = ch_vcf_ann.mix(VCF_ANNOTATE_MERGE.out.vcf_tbi)
-        ch_versions = ch_versions.mix(VCF_ANNOTATE_MERGE.out.versions.first())
+        reports = reports.mix(VCF_ANNOTATE_MERGE.out.reports)
+        vcf_ann = vcf_ann.mix(VCF_ANNOTATE_MERGE.out.vcf_tbi)
+        versions = versions.mix(VCF_ANNOTATE_MERGE.out.versions)
     }
 
     if (tools.split(',').contains('vep')) {
         VCF_ANNOTATE_ENSEMBLVEP(vcf, fasta, vep_genome, vep_species, vep_cache_version, vep_cache, vep_extra_files)
```

Comment:
Would love that, but here, we're just calling the nf-core modules for vep and snpeff, but we're getting closer

Reply:
Allright hit me up when I can help!

Reply:
how much gain is there? I am starting to be more wary of the costs for splitting vs speed gain.

Reply:
It depends on how big the VCF is. For the smaller ones it doesn't make that much of a difference, but it does for the bigger VCFs (e.g. for whole genomes)

Reply:
Also haven't done any proper tests with timestamps etc but has made a difference for those big files

Reply:
I'd expected as such.
Using a spread and gather strategy was always something I considered for the annotation.
VEP is slow because of the huge DB for human, so splitting it up make sense.
Just afraid that we might need to use different intervals than the one we use for variant calling.

Reply:
Yes we currently split on the contigs available in the fasta index. VEP has an option to specify the chromosome(s) that it should annotate for.

Reply:
Also, it's what the ENSEMBL team is recommending: https://github.com/Ensembl/ensembl-vep/tree/main/nextflow

Reply:
Good to know :) Only difference is that we don't split the VCF but use the `--chr` argument (but should be the same result :) )